google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -872,9 +874,13 @@ public void write(Iterable<LogEntry> logEntries, WriteOption... options) {
         logEntries =
             populateMetadata(logEntries, sharedResourceMetadata, this.getClass().getName());
       }
-      // Add partialSuccess option always for request containing instrumentation data
+      // Add partialSuccess = true option always for request which does not have it set explicitly.
+      // For request containing instrumentation data always override it with true.
       writeLogEntries(
-          logEntries, pair.x() ? Instrumentation.addPartialSuccessOption(options) : options);
+          logEntries,
+          pair.x() || writeOptionPartialSuccessFlag == null
```

Comment:
Isn't this missing the case where `writeOptionPartialSuccessFlag == true`?

Reply:
The idea is that if `writeOptionPartialSuccessFlag` present, do not override it since it was explicitly set by user

Reply:
I read it a bit more. Is this equivalent?:

```
add_instrumentation = pair.x()
if (add_instrumentation || writeOptionPartialSuccessFlag == null || writeOptionPartialSuccessFlag == true ) {
    // Add partialSuccess = true option always for request which does not have it set explicitly, or
    // For any request containing instrumentation data
  options = Instrumentation.addPartialSuccessOption(options)
}
writeLogEntries(logEntries, options)
```

If so, I think something like this would make it more readable/maintainable.

I'd suggest possibly moving `addPartialSuccessOption` outside of `Instrumentation` too, since it's now being used by default

Reply:
This logic is designed to **add** partialSuccess when it is not present in `options` and **override** `partialSuccess` with `true` for instrumentation record. I can move `Instrumentation.addPartialSuccessOption(options)` out to common

Reply:
Hmm ok, I see what you're saying. This logic was very hard for me to parse though, would you be open to breaking up the `writeLogEntries` statement into multiple lines? There's a lot happening on one line here that can be hard to keep track of, especially using the `pair.x()` instead of a named variable

Reply:
Let me rewrite for clarity